core/src/validate/validator.rs

```diff
@@ -192,9 +225,26 @@ mod tests {
         let spanish_s2 = AndroidString::localizable("s2", "v'alue %1$d");
         xml_writer::write(&mut spanish_strings.file, vec![spanish_s2.clone()]).unwrap();
 
-        let mut invalid_strings_files = super::validate(res_path.to_str().unwrap())
-            .unwrap()
-            .unwrap_err();
+        let mut invalid_strings_files =
+            super::validate(res_path.to_str().unwrap(), fail_on_unlocalized)
+                .unwrap()
+                .unwrap_err();
+
+        let missing_strings_error_for_fr: Option<missing_strings::MissingStrings>;
+        let missing_strings_error_for_es = if fail_on_unlocalized {
+            missing_strings_error_for_fr = Some(missing_strings::MissingStrings {
+                extra_in_foreign_locale: vec![],
+                extra_in_default_locale: vec![default_s2.clone()],
+            });
+
+            Some(missing_strings::MissingStrings {
+                extra_in_foreign_locale: vec![],
+                extra_in_default_locale: vec![default_s1.clone()],
+            })
+        } else {
+            missing_strings_error_for_fr = None;
+            None
+        };
```

Comment:
what's the reason why this isn't the following:
```
let missing_strings_error_for_fr: Option<missing_strings::MissingStrings>;
let missing_strings_error_for_es: Option<missing_strings::MissingStrings>;
if fail_on_unlocalized {
    missing_strings_error_for_fr = Some(missing_strings::MissingStrings {
        extra_in_foreign_locale: vec![],
        extra_in_default_locale: vec![default_s2.clone()],
    });
    missing_strings_error_for_es = Some(missing_strings::MissingStrings {
        extra_in_foreign_locale: vec![],
        extra_in_default_locale: vec![default_s2.clone()],
    });
} else {
    missing_strings_error_for_fr = None;
    missing_strings_error_for_es = None;
}
```

Reply:
That's what I had initially but `clippy` was complaining that `if..else` should instead be used as an expression

Reply:
Oh okay.
Does it compile if you ignore the warning, or is there a way to silence this particular warning?
I'm not sure I would check this in instead of the original just because clippy was complaining, but that's up to you.

Reply:
The compilation will go through but the CI run would fail. There are ways to silence those. I thought those make this look even uglier 😅. Will try to see if there are other ways to do this